c/types.go

```diff
@@ -81,3 +86,20 @@ type Child struct {
 	cancel      func()
 	wait        func(Shutdown) error
 }
+
+// ChildNotification reports when a child has finished
+type ChildNotification struct {
+	runtimeName string
+	err         error
+}
+
+// RuntimeName returns the runtime name of the child that emitted this exit
+// notification
+func (ce ChildNotification) RuntimeName() string {
```

Comment:
Personal preference, but the getters seem like a bit of an overkill. Here is where I miss scala and immutable values.

Reply:
Yes, I cannot guarantee that if I make `RuntimeName` an attribute, that is not going to be modified somewhere else. 

![is fine](https://thenib.imgix.net/usq/1d97429f-4a64-4d52-bfdb-c36172c05228/this-is-not-fine-001-dae9d5.png?auto=compress,format&cs=srgb&_=dae9d5fc0800f12f5c720be598b6bea6)
Golang is fine